Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/BackoffLogicManagerTest.cs

```diff
@@ -0,0 +1,248 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation
+{
+    using System;
+#if !NET40
+    using System.Diagnostics.Tracing;
+#endif
+
+    using System.Globalization;
+    using System.Net;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.WindowsServer.Channel.Helpers;
+
+#if NET40
+    using Microsoft.Diagnostics.Tracing;
+#endif
+
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Channel.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Web.TestFramework;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Helpers;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation;
+    
+    using Assert = Xunit.Assert;
+
+    public class BackoffLogicManagerTest
+    {
+        [TestClass]
+        public class GetBackendResponse
+        {
+            public void ReturnNullIfArgumentIsNull()
+            {
+                var manager = new BackoffLogicManager(TimeSpan.Zero);
+                Assert.Null(manager.GetBackendResponse(null));
+            }
+
+            public void ReturnNullIfArgumentEmpty()
+            {
+                var manager = new BackoffLogicManager(TimeSpan.Zero);
+                Assert.Null(manager.GetBackendResponse(string.Empty));
+            }
+
+            public void IfContentCannotBeParsedNullIsReturned()
+            {
+                var manager = new BackoffLogicManager(TimeSpan.Zero);
+                Assert.Null(manager.GetBackendResponse("ab}{"));
+            }
+
+            public void IfContentIsUnexpectedJsonNullIsReturned()
+            {
+                var manager = new BackoffLogicManager(TimeSpan.Zero);
+                Assert.Null(manager.GetBackendResponse("[1,2]"));
+            }
+
+            public void BackendResponseIsReturnedForCorrectContent()
+            {
+                string content = BackendResponseHelper.CreateBackendResponse(100, 1, new[] {"206"}, 84);
+
+                var manager = new BackoffLogicManager(TimeSpan.Zero);
+                var backendResponse = manager.GetBackendResponse(content);
+
+                Assert.Equal(1, backendResponse.ItemsAccepted);
+                Assert.Equal(100, backendResponse.ItemsReceived);
+                Assert.Equal(1, backendResponse.Errors.Length);
```

Comment:
I think this might be confusing or misleading for someone else reading the code in future because if 100 items are sent and only 1 is accepted then wouldn't the length of the errors be 99?